### PR TITLE
[SDXL Flax] fix SDXL flax init

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Make sure Flax init is correct for different model sizes

### CODE REVIEW

File: src/diffusers/models/unet_2d_condition_flax.py [hunk 1]
```diff
@@ -134,8 +134,13 @@ def init_weights(self, rng: jax.random.KeyArray) -> FrozenDict:
...
         added_cond_kwargs = None
         if self.addition_embed_type == "text_time":
-            # TODO: how to get this from the config? It's no longer cross_attention_dim
-            text_embeds_dim = 1280
+            # we retrieve the expected `text_embeds_dim` by first checking if the architecture is a refiner
+            # or non-refiner architecture and then by "reverse-computing" from `projection_class_embeddings_input_dim`
+            is_refiner = 5 * self.config.addition_time_embed_dim + self.config.cross_attention_dim == self.config.projection_class_embeddings_input_dim
+            num_micro_conditions = 5 if is_refiner else 6
+
+            text_embeds_dim = self.config.projection_class_embeddings_input_dim - (num_micro_conditions * self.config.addition_time_embed_dim)
```
Comment (patrickvonplaten) [inline src/diffusers/models/unet_2d_condition_flax.py:142]: 2816 - 6 * 256 = 1280 for base and 2560 - 5 * 256 = 1280 for refiner

File: src/diffusers/pipelines/stable_diffusion_xl/pipeline_flax_stable_diffusion_xl.py [hunk 2]
```diff
@@ -215,14 +215,15 @@ def _generate(
...
-        # scale the initial noise by the standard deviation required by the scheduler
-        latents = latents * params["scheduler"].init_noise_sigma
 
         # Prepare scheduler state
         scheduler_state = self.scheduler.set_timesteps(
             params["scheduler"], num_inference_steps=num_inference_steps, shape=latents.shape
         )
 
+        # scale the initial noise by the standard deviation required by the scheduler
+        latents = latents * scheduler_state.init_noise_sigma
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/stable_diffusion_xl/pipeline_flax_stable_diffusion_xl.py:225]: We need to set the init_noise_sigma atfter scaling

**Comment by patrickvonplaten** [comment]
@pcuenca I'm now getting almost identical results on CPU with PyTorch vs. Flax for dummy inputs:

**Flax**
```py
from diffusers import FlaxStableDiffusionXLPipeline
import numpy as np
import jax.numpy as jnp
import jax

path = "hf-internal-testing/tiny-stable-diffusion-xl-pipe"

pipe, params = FlaxStableDiffusionXLPipeline.from_pretrained(path)

prompt = "An astronaut riding a green horse on Mars"
negative_prompt = "ugly"
steps = 3

batch_size, height, width, ch = 1, 32, 32, 4
num_elems = batch_size * height * width * ch
rng = jax.random.PRNGKey(0)
latents = (jnp.arange(num_elems) / num_elems)[:, None, None, None].reshape(batch_size, ch, width, height)

print("latents", np.abs(np.asarray(latents)).sum())

prompt_embeds = pipe.prepare_inputs(prompt)
neg_prompt_ids = pipe.prepare_inputs(negative_prompt)

image = pipe(prompt_embeds, params, rng, neg_prompt_ids=neg_prompt_ids, latents=latents, num_inference_steps=3, output_type="np").images[0]

print(np.abs(np.asarray(image)).sum())
```

**PT**
```py
import torch
import numpy as np
from diffusers import StableDiffusionXLPipeline

path = "hf-internal-testing/tiny-stable-diffusion-xl-pipe"

pipe = StableDiffusionXLPipeline.from_pretrained(path)
pipe.unet.set_default_attn_processor()

prompt = "An astronaut riding a green horse on Mars"
neg_prompt = "ugly"
steps = 3

batch_size, height, width, ch = 1, 32, 32, 4
num_elems = batch_size * height * width * ch
latents = (torch.arange(num_elems) / num_elems)[:, None, None, None].reshape(batch_size, ch, width, height)
print("latents", latents.abs().sum())

image = pipe(prompt, negative_prompt=neg_prompt, latents=latents, num_inference_steps=3, output_type="np", guidance_scale=7.5).images[0]

print(np.abs(image).sum())
```

Getting:
**PT**: 6237.967
**Flax**: 6237.9585

File: src/diffusers/pipelines/stable_diffusion_xl/pipeline_flax_stable_diffusion_xl.py (hunk 2 above)
Comment (pcuenca) [reply src/diffusers/pipelines/stable_diffusion_xl/pipeline_flax_stable_diffusion_xl.py:225]: Oh yes, great catch!

**Pull request merged**